pkg/controller/machine/kubeconfig.go

```diff
@@ -43,6 +55,24 @@ func (c *Controller) createBootstrapKubeconfig(name string) (*clientcmdapi.Confi
 	return outConfig, nil
 }
 
+func (c *Controller) getTokenFromServiceAccount(name types.NamespacedName) (string, error) {
+	sa, err := c.kubeClient.CoreV1().ServiceAccounts(name.Namespace).Get(name.Name, metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+	for _, serviceAccountSecretName := range sa.Secrets {
+		serviceAccountSecret, err := c.kubeClient.CoreV1().Secrets(name.Namespace).Get(serviceAccountSecretName.Name, metav1.GetOptions{})
+		if err != nil {
+			return "", err
+		}
+		if serviceAccountSecret.Type != v1.SecretTypeServiceAccountToken {
+			continue
+		}
+		return string(serviceAccountSecret.Data["token"]), nil
```

Comment:
There's actually a const `ServiceAccountTokenKey = "token"` in (`k8s.io/api/core/v1`)

Reply:
Nice, thanks for the hint, updated

/hold cancel